libcaf_core/caf/scheduled_actor.cpp

```diff
@@ -231,15 +231,27 @@ bool scheduled_actor::cleanup(error&& fail_state, execution_unit* host) {
   // Clear state for open requests.
   awaited_responses_.clear();
   multiplexed_responses_.clear();
-  // Cancel any active flow.
+  // Cancel flows and streams.
+  stream_sources_.clear();
+  if (!stream_subs_.empty()) {
+    decltype(stream_subs_) subs;
+    subs.swap(stream_subs_);
```

Comment:
To better understand this: why did you swap out here instead of clearing in-place? Can `ptr->cancel()` index into `stream_subs_`? Same question for below code with bridges and disposables.

Reply:
Exactly. Calling `cancel`/`drop` might try to erase itself from the map. Hence, we must make sure we're not changing the map while iterating it.